src/pages/workspaces/workspace/Data.js

```diff
@@ -194,6 +202,72 @@ class WorkspaceData extends Component {
     ])
   }
 
+  renderDownloadButton() {
+    const { namespace, name } = this.props
+    const { selectedDataType, orchestrationRoot } = this.state
+
+    return form({
+      style: { display: 'inline' },
+      method: 'POST',
+      action: `${orchestrationRoot}/cookie-authed/workspaces/${namespace}/${name}/entities/${selectedDataType}/tsv`
+    }, [
+      input({ type: 'hidden', name: 'FCtoken', value: auth.getAuthToken() }),
+      /*
+       * TODO: once column selection is implemented, add another hidden input with name: 'attributeNames' and
+       * value: comma-separated list of attribute names to support downloading only the selected columns
+       */
+      tooltip({
+        component: buttonPrimary({ type: 'submit', disabled: !orchestrationRoot }, [
+          icon('download', { style: { marginRight: '0.5rem' } }),
+          'Download'
+        ]),
+        text: 'Download all data as a file',
+        arrow: 'center', align: 'center'
+      })
+    ])
+  }
+
+  renderCopyButton() {
+    const { entities, selectedDataType, entityMetadata, copying, copied } = this.state
+
+    return h(Fragment, [
+      tooltip({
+        component: buttonPrimary({
+          style: { margin: '0 1rem' },
+          onClick: async () => {
+            const attributeNames = entityMetadata[selectedDataType].attributeNames
+
+            const entityToRow = entity =>
+              _.join('\t', [
+                entity.name, ..._.map(
+                  attribute => Utils.entityAttributeText(entity.attributes[attribute]),
+                  attributeNames)
+              ])
+
+            const header = _.join('\t', [`${selectedDataType}_id`, ...attributeNames])
+
+            const str = _.join('\n', [header, ..._.map(entityToRow, entities)]) + '\n'
+
+            try {
+              this.setState({ copying: true })
+              await navigator.clipboard.writeText(str)
```

Comment:
This is the "highly experimental" thing that works in Chrome and possibly nowhere else.

Reply:
What about [this](https://hackernoon.com/copying-text-to-clipboard-with-javascript-df4d4988697f)?

Reply:
Or potentially even easier, [this](https://clipboardjs.com)

Reply:
First one is way too complicated. Second one only copies from an element.

Reply:
This looks promising, and is based on emerging standards. https://github.com/lgarron/clipboard-polyfill